src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -1936,61 +1950,152 @@ namespace FanCoilUnits {
+				Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutMaxC );
+				if ( ! ColdFlowLocked ) {
+					FanCoil( FanCoilNum ).QUnitOutMaxC = QUnitOutMaxC;
+				} else {
+					QUnitOutMaxC = FanCoil( FanCoilNum ).QUnitOutMaxC;
+					MdotLockC = mdot; // save locked flow
+				}
+				QZnReq = ZoneSysEnergyDemand( ZoneNum ).RemainingOutputReqToCoolSP;
+				if ( QUnitOutMaxC < QZnReq ) {
 					// more cooling than required, find reduced water flow rate to meet the load
-					ControlNode = FanCoil( FanCoilNum ).ColdControlNode;
-					ControlOffset = FanCoil( FanCoilNum ).ColdControlOffset;
-					MaxWaterFlow = FanCoil( FanCoilNum ).MaxColdWaterFlow;
-					MinWaterFlow = FanCoil( FanCoilNum ).MinColdWaterFlow;
-					//On the first HVAC iteration the system values are given to the controller, but after that
-					// the demand limits are in place and there needs to be feedback to the Zone Equipment
-					if ( ! FirstHVACIteration ) {
-						MaxWaterFlow = Node( ControlNode ).MassFlowRateMaxAvail;
-						MinWaterFlow = Node( ControlNode ).MassFlowRateMinAvail;
+					//solve for the cold water flow rate with no limit set by flow rate lockdown
+					Par( 1 ) = double( FanCoilNum );
+					Par( 2 ) = 0.0; // FLAG, IF 1.0 then FirstHVACIteration equals TRUE, if 0.0 then FirstHVACIteration equals false
+					if ( FirstHVACIteration ) Par( 2 ) = 1.0;
+					Par( 3 ) = ControlledZoneNum;
+					Par( 4 ) = QZnReq;
+					SolveRegulaFalsi( 0.001, MaxIterCycl, SolFlag, CWFlow, CalcFanCoilCWLoadResidual, 0.0, MaxWaterFlow, Par );
+					if ( SolFlag == -1 ) {
+						ShowWarningError( "Cold Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name );
+						ShowContinueError( "  Iteration limit exceeded in calculating water flow rate " );
+						ShowRecurringWarningErrorAtEnd( "Cold water flow Iteration limit exceeded in fan coil unit " + FanCoil( FanCoilNum ).Name, FanCoil( FanCoilNum ).MaxIterIndexH );
+					}
+					else if ( SolFlag == -2 ) {
+						ShowWarningError( "Cold Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name );
+						ShowContinueError( "  Bad cold water mass flow limits" );
+						ShowRecurringWarningErrorAtEnd( "Cold Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name, FanCoil( FanCoilNum ).MaxIterIndexC );
+					}
+				} else {
+					// demand greater than capacity
+					CWFlow = MaxWaterFlow;
+				}
+				if ( ! ColdFlowLocked ) {
+					mdot = CWFlow; // not flowlocked - set flow to CWFlow
+					SetComponentFlowRate( mdot, FanCoil( FanCoilNum ).ColdControlNode, FanCoil( FanCoilNum ).ColdPlantOutletNode, FanCoil( FanCoilNum ).CWLoopNum,
+						FanCoil( FanCoilNum ).CWLoopSide, FanCoil( FanCoilNum ).CWBranchNum, FanCoil( FanCoilNum ).CWCompNum );
+					Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOut ); // get QUnitOut
+				}
+				else {
+					// flow lock on 
+					if ( MdotLockC > CWFlow ) { // if mdot > CWFlow, bypass extra flow
```

Comment:
This is the kludge that fixes the bug. In the defect file (unreproducible elsewhere) plant flows are getting locked down due to flow imbalance in plant between supply and demand. This is a large flow imbalance and I could find no reason for it. With the flow locked, the constant fan variable flow fan coil can't respond to the load. So I stuck in an internal bypass - the flow through the coil can still vary but the total flow is constant at the locked value. The bypassed water rejoins at the outlet node.